### PR TITLE
3.0 empty translations entity fix

### PR DESCRIPTION
**Pull request by robertpustulka**
Refs #6833

I came up with this simple solution.

I've also switched the instanceof check to `EntityInterface` instead of `Entity` class.

### CODE REVIEW

File: src/ORM/Behavior/Translate/TranslateTrait.php [hunk 1]
```diff
@@ -46,8 +46,10 @@ public function translation($language)
             $created = true;
         }
 
-        if ($created || empty($i18n[$language]) || !($i18n[$language] instanceof Entity)) {
-            $i18n[$language] = new Entity();
+        if ($created || empty($i18n[$language]) || !($i18n[$language] instanceof EntityInterface)) {
+            $className = get_class($this);
+
+            $i18n[$language] = new $className;
```
Comment (dereuromark) [inline src/ORM/Behavior/Translate/TranslateTrait.php]: As per CS this might need (): `$className()`.

Comment (robertpustulka) [inline src/ORM/Behavior/Translate/TranslateTrait.php]: @dereuromark travis seems to be happy :)

Comment (dereuromark) [inline src/ORM/Behavior/Translate/TranslateTrait.php]: Well, doesn't mean we have a sniff for it, but usually all class inits have the trailing `()`.

**Comment by markstory** [comment]
Looks good once the extra() are included.

**Comment by robertpustulka** [comment]
@markstory done

**Pull request merged**